x-pack/functionbeat/magefile.go

```diff
@@ -74,13 +115,8 @@ func TestPackages() error {
 }
 
 // Update updates the generated files (aka make update).
-func Update() error {
-	return sh.Run("make", "update")
-}
-
-// Fields generates a fields.yml for the Beat.
-func Fields() error {
-	return devtools.GenerateFieldsYAML()
+func Update() {
```

Comment:
I would like to see each project have an "update" namespace (e.g. `mage update:fields` or `mage update:docs`) that's tied in with the common targets for `check` and `docs`. This is adapted from `winlogbeat/scripts/mage/update.go` and you would need plug in your logic for some of the targets. 

```
import (
	"github.com/magefile/mage/mg"

	devtools "github.com/elastic/beats/dev-tools/mage"
	"github.com/elastic/beats/dev-tools/mage/target/common"
	"github.com/elastic/beats/dev-tools/mage/target/docs"
)

func init() {
	common.RegisterCheckDeps(Update.All)

	docs.RegisterDeps(Update.FieldDocs)
}

// Update target namespace.
type Update mg.Namespace

var Aliases = map[string]interface{} {
	"update":    Update.All,
}

// All updates all generated content.
func (Update) All() {
	mg.Deps(Update.Fields, Update.Config, Update.FieldDocs)
}

// Config updates the Beat's config files.
func (Update) Config() error {
	return config()
}

// Fields updates all fields files (.go, .yml).
func (Update) Fields() {
	mg.Deps(fb.All)
}

// FieldDocs updates the field documentation.
func (Update) FieldDocs() error {
	mg.Deps(fb.FieldsAllYML)
	return devtools.Docs.FieldDocs(devtools.FieldsAllYML)
}
```